backend/assignments/1/src/main/java/org/example/CoinList.java

```diff
@@ -0,0 +1,131 @@
+package org.example;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+
+public class CoinList {
+
+    private static final LoggerService log = new LoggerService();
+    public  List<Coin> coinListArr = new CopyOnWriteArrayList<>();
+
+    private TraderList traderList;
+    public CoinList(TraderList traderList) {
+        this.traderList = traderList;
+    }
+
+    public  synchronized void updatePrice(Coin coin, double price){
```

Comment:
What if user1 needs to work on Coin1 and user2 on coin2.
But user2 will still be in wait state untill the method reopoens lock.
Can the locking be in a better place?